### ISSUE TITLE
cancelling the archive check doesnt kill the underlying borg command

### ISSUE THREAD

**Issue by sandrotosi** [borgbase/vorta#930]
**Describe the bug**
cancelling the archive check doesnt kill the underlying borg command

**To Reproduce**
1. have a backup enabled with the option "Validate repository data every ..." enabled
2. when the check step is running, click cancel at the bottom right
3. the action is marked as cancelled but the underlying `borg`  command is not killed and keeps running.

**Environment (please complete the following information):**
- OS: Debian unstable
- Vorta version: 0.7.5
- Installed from: debian package

**Comment by m3nu** [comment]
Wow. We are already *very* aggressive in killing that process. Basically if it doesn't react to SIGINT within X seconds, the whole process group is terminated. So a bit surprising.

**Comment by m3nu** [comment]
Is this a remote or local repo?

**Comment by sandrotosi** [comment]
this is a "local" repo, as in it's an NFS share mounted on a local directory

**Comment by m3nu** [comment]
Tried to reproduce this on a VMWare-mounted share, but failed. Maybe it needs a slow network share to reproduce.

<img src="https://user-images.githubusercontent.com/3916435/114202414-bcadad00-9989-11eb-8e0b-b29333670b18.jpg" width="250" />

**Comment by sandrotosi** [comment]
i wouldnt call this a slow network share: it's a 1gbps ethernet cabled connection from my laptop, to a switch, to a nas.

FTR it happened again a couple of days ago (that's the weekly check running *not* the backup): i decided to cancel the check but the process didnt die, so i had to kill it by hand

**Comment by m3nu** [comment]
I also tried cancelling, when running the `borg check` command to a remote repo over the internet.

Will cancelling also fail for you for remote repos or just network shares?

**Comment by m3nu** [comment]
Here the "kill code", if you want to play with it: https://github.com/borgbase/vorta/blob/master/src/vorta/borg/borg_thread.py#L277

**Comment by sandrotosi** [comment]
> I also tried cancelling, when running the `borg check` command to a remote repo over the internet.

ok this is interesting: did you go to the "Archives" tab and press "Check" at the bottom of it? i was able to cancel the check by following that sequence of steps.

The situation for which i reported this bug is slightly different: i cannot kill the check when executed automatically after a backup, as scheduled by enabling "Validate repository data ever X weeks" in the "Schedule" tab.

**Comment by m3nu** [comment]
Oh. I started it manually from *Archives*.

The scheduled check will use the same Borg thread, but there could be small differences. So let me try that too.

**Comment by sandrotosi** [comment]
leaving a comment to please the stale bot: this is still happening to me (which is no surprise since i didnt upgrade vorta :) )

**Comment by sandrotosi** [comment]
please reopen, this is still valid

**Comment by sandrotosi** [comment]
ping

**Comment by sandrotosi** [comment]
anothet ping; can the stale bot be disabled on this issue?

**Comment by m3nu** [comment]
Does it still happen? This part changed in major ways in 0.8.x

**Comment by sandrotosi** [comment]
looks like we still have 0.7.x in debian -- @sten0 can you upgrade to the latest vorta release?

**Comment by m3nu** [comment]
Already working on it. 😇 #1133

**Comment by sten0** [comment]
Hi @sandrotosi,

I have 0.8.2 in git, but unittests are completely failing...I'm not sure what's going on, and I haven't been able to make progress in any of the 20min sprints I've attempted (limited spaces of free time).  In particular, the Peewee errors concern me, and the failure appears to be identical between Peewee 3.14.4 and 3.14.8.  At this point I suspect something changed in the Debian dependency chain or infrastructure, but I haven't been able to successfully identify what it might be.

Hi @m3nu, looking at the git log, I didn't see anything that would have broken all tests.  Maybe I missed something?  If Vorta is at fault, my best guess is that it's due use and testing against an obsolete Peewee (given the error output), but I'm not sure why 0.7.8 was unaffected with Python 3.9 and  Peewee 3.14.4:
```
flatpak/dependencies/python3-peewee.json
2:    "name": "python3-peewee",
5:        "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} peewee"
10:            "url": "https://files.pythonhosted.org/packages/90/58/04bc705057694cac14c5c9a00dc83a80aafa1de27785227d3ee245cd4165/peewee-3.11.2.tar.gz",
```

I've attached a build log, because it looks like I won't be able to resolve this issue in a timely fashion without another set of eyes
[vorta_0.8.2-1_amd64-2021-12-22T19:16:06Z.txt](https://github.com/borgbase/vorta/files/7764871/vorta_0.8.2-1_amd64-2021-12-22T19.16.06Z.txt)

**Comment by sandrotosi** [comment]
> I have 0.8.2 in git, but unittests are completely failing...I'm not sure what's going on, and I haven't been able to make progress in any of the 20min sprints I've attempted (limited spaces of free time). In particular, the Peewee errors concern me, and the failure appears to be identical between Peewee 3.14.4 and 3.14.8. At this point I suspect something changed in the Debian dependency chain or infrastructure, but I haven't been able to successfully identify what it might be.

i gave it a quick look yday, a couple of things that needs fixing are:

1. mkdir -p $XDG_RUNTIME_DIR (at lease in a pbuilder chroot), else dbus-service doesnt run
2. quilt package 0006-* needs to be extended to check if the desktop file is available locally (ie we're running the test at build time) or only globally (meaning the code is running at install time)

that didnt make much of a change, as tests still fail, so i got annoyed and moved onto something else :)

**Comment by m3nu** [comment]
We re-arranged some DB-related files, but no other large items changed.

I know how to run Debian tests locally and will find the issue.

**Comment by m3nu** [comment]
Just set up Debian Unstable (bookworm) and ran Vorta using only Debian's own packages under Xfce. No problems with the tests. Here the dependencies I used. When looking at the [control file](https://salsa.debian.org/python-team/packages/vorta/-/blob/debian/master/debian/control), some deps could be removed I think.

Next I will try to set up sbuild and see what needs changing exactly on the package.

```
 # app deps only
apt install python3-appdirs python3-paramiko python3-pyqt5 python3-peewee python3-secretstorage borgbackup

# test deps
apt install python3-pytest python3-pytest-mock python3-pytestqt

# Install to virtual env, but allow system packages
python3 -m venv --system-site-packages ~/env

# Install vorta. Note that no dependencies are installed.
source ~/env/bin/activate && cd vorta && pip install -e vorta

# run tests, no error (while running minimal xfce)
python3 -m pytest
```
![Screen Shot 2021-12-23 at 19 52 48](https://user-images.githubusercontent.com/3916435/147264172-80d097e7-8dd1-42df-9ec2-6d37ca1149e4.jpg)

**Comment by m3nu** [comment]
Happy to report that I found the root cause of the failing tests: Seems like the test tool would set a new home directory, which didn't have a `~/.config` folder yet, which we use for autostart settings. With this parent folder missing, creating our `~/.config/autostart` also failed. So not related to Peewee really.

With a minor change to `Path.mkdir()` parent folders are created and tests pass. I'll do a small PR shortly. Tested with (on amd64): `pybuild --test-pytest` and `sbuild`. All mostly passing.

Next steps/open points:

1. Fix `Path.mkdir()` issue causing all tests to fail: done in #1146 
2. Remove unused `setuptools_git` (project not updated for 5 years) dependency from Vorta: done in #1147 
3. Some unused dependencies could be removed from your [package](https://salsa.debian.org/python-team/packages/vorta). Should I do a clean PR on Salsa? In detail:
  - Should be safe to remove: `python3-{keyring,networkmanager,mock,pytest-runner,setuptools-git}`
  - For those I'm not fully sure what they're used for: `python3-pyqt5.qsci`, `qttools5-dev-tools`
  - To add explicitly: `libqt5svg5` (really needed for our SVG icons)
4. I'm still seeing one test fail and some hang (using dbus) while in sbuild. Maybe I'm not using it properly. Let's see what you get.

Merry Christmas everyone! 🎄🎁

**Comment by m3nu** [comment]
Here my PR for the Debian packaging repo (my first). Please let me know if those changes make sense or if anything could be improved. I was a bit confused by the repeated dependencies in `tests/control` and `control`. [Here](https://ci.debian.net/doc/file.TUTORIAL.html) they used `@builddeps@` to refer to build deps in tests. Not sure what's the right approach.

https://salsa.debian.org/python-team/packages/vorta/-/merge_requests/1

**Comment by m3nu** [comment]
I've now tested my proposed packaging changes with the current Vorta `master` branch and `sbuild` runs without errors. Let's look into it in detail after the holidays.

**Comment by sten0** [comment]
Hi m3nu!

Would you please share your vorta*.build and vorta*.buildinfo files?  I've been unable to replicate your success.

Thanks,
Nicholas

**Comment by sten0** [comment]
> Here my PR for the Debian packaging repo (my first).

Congrats on your first Debian contribution, and thank you for the help :-)

> Please let me know if those changes make sense or if anything could be improved

Replied on salsa.

> I was a bit confused by the repeated dependencies in `tests/control` and `control`. [Here](https://ci.debian.net/doc/file.TUTORIAL.html) they used `@builddeps@` to refer to build deps in tests. Not sure what's the right approach.
>

There are three levels of tests:  build-time tests (tests compilation and unit tests), reprobuild (tests reproducibility of build-time tests and reproducibility of the generated binary package), and autopkgtests (tests the package as-installed at a system level on a clean Debian installation).  Tests/control is for autopkgtests, and debian/control is for build-time and reprobuild tests.

Search for `@builddeps@` at https://people.debian.org/~eriberto/README.package-tests.html.  I chose to explicitly list them because I want the binary package to fail early on a clean install in case of missing dependencies; this is rare, but I've observed periodic bugs with Python dependency resolution -> runtime dependencies in other software.  Doing it this way is more work than the variable/macro (I didn't check which), but is less work than manually testing 
 
Happy New Year!

**Comment by sandrotosi** [comment]
please refer to the up-to-date location for autopkgtest here: https://salsa.debian.org/ci-team/autopkgtest/-/tree/master/doc (eriberto's page is outdate/unmaintained)

**Comment by m3nu** [comment]
Thanks for the background, @sten0 and @sandrotosi. And thanks for the updates. All this Debian packaging is really another world. 😄

Agreed that fixing tests is the main priority. I thought I had done that first. Those changes went into Vorta's repo.

Good news is that the test issue seems really minor and related to `0006-fix-desktop_file-path-in-tests.patch`. My tests fail due to the file mentioned there missing. When I comment the patch out in `debian/patches/series` it all works.

Here the steps I did. Hopefully I didn't go wrong on the way. This was with a new project folder and fully updated *unstable* install:

```
debcheckout -a vorta && cd vorta
gbp pull --redo-pq
# comment out 0006-fix-desktop... in /debian/patches/series
gbp buildpackage --git-ignore-new
sbuild  # runs successfully
```

Here the successful build files you requested on Salsa (moving the conversation here to have it all in the same place):
- [vorta_0.8.2-1_amd64.buildinfo.gz](https://github.com/borgbase/vorta/files/7923059/vorta_0.8.2-1_amd64.buildinfo.gz)
- [vorta_0.8.2-1_amd64.build.gz](https://github.com/borgbase/vorta/files/7923060/vorta_0.8.2-1_amd64.build.gz)

To move forward:
- I'm not sure where the `.desktop` file in this `0006` patch comes from, but its absence seems to make the tests fail. Maybe Debian would usually automatically put it in the right place, but doesn't for some reason.
- The error related to `setuptools_git` seen in your last build log went away. From what I see in the same build log, this build still used the old patch set and didn't remove the `setuptools_git` dependency.
- Once we have confirmed that the issue is really with the `.desktop` file, I'd still like to remove old dependencies before they cause trouble. I'll do a new PR on Salsa for that after testing.

<img width="300" src="https://user-images.githubusercontent.com/3916435/150728326-1581d324-8fe5-4cbd-9199-674efae1335a.jpg" />

**Comment by sten0** [comment]
Hi Manu!

Manu ***@***.***> writes:

> Thanks for the background, @sten0 and @sandrotosi. And thanks for the updates. All this Debian packaging is really another world. 😄
>

I hear you :-)  It's off topic for this issue, but I'd be curious to
read about your impressions and experiences.

Spoiler: I'm surprised the desktop file was the central point of
failure, because that's highly unusual for *NIX software.  Different
assumptions from working on different OSes, you know?  I confess that I
didn't see it as that significant, and that this bias got in the way.
More on this later.

> Agreed that fixing tests is the main priority. I thought I had done that first. Those changes went into Vorta's repo.
>

Sorry it took me so long to push my work and cherry pick additional
fixes.  It's totally fair to expect me to do that and to focus on the
upstream Vorta repo.  If you'd like, feel free to @sten0 to signal me to
cherry pick high priority targeted fixes.

> Good news is that the test issue seems really minor and related to `0006-fix-desktop_file-path-in-tests.patch`. My tests fail due to the file mentioned there missing. When I comment the patch out in `debian/patches/series` it all works.
>

Wow, good find!  Once again I'm shocked that this was the central point
of failure.  More on this later.

> Here the steps I did. Hopefully I didn't go wrong on the way. This was
> with a new project folder and fully updated *unstable* install:
[snip]

Thank you for the build log and buildinfo.  I skimmed it, it looks
good to me, and I'm optimistic this is the case.

> To move forward:
> - I'm not sure where the `.desktop` file in this `0006` patch comes
> from, but its absence seems to make the tests fail. Maybe Debian would
> usually automatically put it in the right place, but doesn't for some
> reason.

I added this patch to fix the autopkgtest (CI) case.  As Sandro noted
some time ago, it needs to be fixed for the build-time tests to pass.  I
hope he'll push his work soon, since he found the issue and fixed it,
and because the best test I can think of is checking the value of
$AUTOPKGTEST_TMP envvar, and then using the system-wide location if it's
set, else using the unpatched path.  The build-time tests passed with the
0.7.x series, so I'm guessing the importance of the .desktop file was
increased for the 0.8.x series, maybe in one of those big "refactoring"
commits?

> - The error related to `setuptools_git` seen in your last build log
> went away. From what I see in the same build log, this build still
> used the old patch set and didn't remove the `setuptools_git`
> dependency.

I think this happened because while you were working on your salsa PR I
was importing 0010-Remove-unused-setuptools-git-dependency.patch ;-)
Once again, I'm sorry for forgetting to push for so long; I imagine this
made things confusing and not nice.

> - Once we have confirmed that the issue is really with the `.desktop` file, 

I can confirm that disabling 0006-fix-desktop_file-path-in-tests.patch
fixes build-time unit tests, which means the logic branch that Sandro
worked on will work.  Sandro would you please push your work when you
have a free moment?

> I'd still like to remove old dependencies before they cause
> trouble. I'll do a new PR on Salsa for that after testing.
>

I have no objection to removing them for decrufting/aesthetic reasons,
but it will be a problem if they could cause trouble, because Debian
packages are system-level (not confined to a virtualenv nor container).
A user might manually install these packages for something unrelated to
Vorta...  A workaround would be to declare a Conflict in the Debian
package, but that wouldn't be a good solution.

If the old dependencies could cause trouble then that would be a reason
to continue to install them to the build environment, to test if they
cause trouble, because it's better for the build to fail than for a user
to user to encounter a bug that may result in a failed backup, and thus
data loss when attempting to recover from hardware failure.

> <img width="300" src="https://user-images.githubusercontent.com/3916435/150728326-1581d324-8fe5-4cbd-9199-674efae1335a.jpg" />
>

Yay, finally! :-D

**Comment by sten0** [comment]
And thank you Manu and Sandro, I really appreciate the help!

**Comment by sandrotosi** [comment]
> Sandro would you please push your work when you have a free moment?

i'm a bit uncertain what work you want me to push exactly? i dont think i produced any change (either for upstream or the debian package) that needs pushing

**Comment by sten0** [comment]
Sandro Tosi ***@***.***> writes:

>> Sandro would you please push your work when you have a free moment?
>
> i'm a bit uncertain what work you want me to push exactly? i dont think i produced any change (either for upstream or the debian package) that needs pushing
>

Previously you wrote:

> i gave it a quick look yday, a couple of things that needs fixing are:
>
> 1. mkdir -p $XDG_RUNTIME_DIR (at lease in a pbuilder chroot), else dbus-service doesnt run
> 2. quilt package 0006-* needs to be extended to check if the desktop file is available locally (ie we're
> running the test at build time) or only globally (meaning the code is running at install time)
>
> that didnt make much of a change, as tests still fail, so i got annoyed and moved onto something else :)

Since you identified the problem in
0006-fix-desktop_file-path-in-tests.patch and already did the work, I
feel like you should take the credit :-)  Otherwise I'll modify the patch
to check for $AUTOPKGTEST_TMP, and then use the /usr/….desktop (existing
logic in the patch) for that case, else use the upstream location.  I
also wonder if you used a better method for your fix, or if it would be
better to fall back to probing for the file rather than using that envvar.

I was surprised sbuild didn't fail due to issue 1, so maybe it's
pbuilder specific?  If so, then I don't know the correct way to check
for a pbuilder environment.  Or maybe I was optimistic and tired and
missed something.  'not sure, I'll get some sleep and check later.

Cheers,
Nicholas

**Comment by m3nu** [comment]
> If the old dependencies could cause trouble then that would be a reason to continue to install them to the build environment, to test if they cause trouble, because it's better for the build to fail than for a user to user to encounter a bug that may result in a failed backup, and thus data loss when attempting to recover from hardware failure.

I didn't mean that their **presence** will cause trouble. 😄 That's not an issue of course. Just that we don't want to drag along unmaintained packages.

And good to hear things are moving. Anything that I should do except to flag old dependencies and release 0.8.3?

**Comment by sandrotosi** [comment]
> Sandro Tosi ***@***.***> writes:
> > Sandro would you please push your work when you have a free moment? i'm a bit uncertain what work you want me to push exactly? i dont think i produced any change (either for upstream or the debian package) that needs pushing
> Previously you wrote:
> i gave it a quick look yday, a couple of things that needs fixing are: 1. mkdir -p $XDG_RUNTIME_DIR (at lease in a pbuilder chroot), else dbus-service doesnt run 2. quilt package 0006-* needs to be extended to check if the desktop file is available locally (ie we're running the test at build time) or only globally (meaning the code is running at install time) that didnt make much of a change, as tests still fail, so i got annoyed and moved onto something else :)

that was it: that message *was* the work and i have no code to commit

> Since you identified the problem in 0006-fix-desktop_file-path-in-tests.patch and already did the work, I feel like you should take the credit :-)

i have 0 interest for credit, feel free to make all the changes that you think are needed, using or not what i wrote before

**Comment by sten0** [comment]
@Manu  Fiouf @ what you meant! :-)

Before tagging 0.8.3, I'd migrate the flatpak to python3-llfuse 1.4.1
(bugfix release).

For the Debian side of things, solving the failure of
`test_archive_mount`, when the `os` library is imported in autostart.py,
is needed.  I need `os` to either check if the desktop file is in /usr
(for autopkgtests/CI), or alternatively I can use os.environ to test for
AUTOPKGTEST_TMP.  Could this failure be mocking or
monkeypatching-related?  Honestly, I'm not sure, except that I don't
think that it's a simple timeout (as the error indicates) because 20
seconds should be more than enough.  See today's new commits on
debian/master to reproduce (HEAD @commit:0180782).  I wonder if #1171
could be related?

Meanwhile, I still worry that there's something funny in the autostart
code, eg: #1152.  Could 4c3e97a have introduced a partial regression?

If possible, it would also be nice to see the null images handled
(during unit tests), because it makes reading the logs unpleasant eg:

  QImage::scaleHeight: Image is a null image
  …repeats another 16 times before/after each unit test

https://stackoverflow.com/questions/21154817/qimagescaleheight-transformation-mode-returning-null-image

@sandrotosi, sure, as you prefer!  I'll continue to plod away, but hit
the blocker noted above.  I'm also wondering if it will be a good idea
to set $HOME and $XDG_RUNTIME dir to /tmp/{home,run}, respectively (done
today on Salsa), which is a stable path, and then attempt to reenable
the disabled test[s]...  In due time, I mean.  I'm certain you'll agree
that that, and the pbuilder case are lower priority than fixing FTBFS on
buildd ;-)

**Comment by m3nu** [comment]
> Before tagging 0.8.3, I'd migrate the flatpak to python3-llfuse 1.4.1 (bugfix release).

On it.

> For the Debian side of things, solving the failure of `test_archive_mount`, when the `os` library is imported in autostart.py, is needed.

Will try to reproduce and see what happens. #1171 is a Borg error. Wrong repo. Still need to reply later.

> Meanwhile, I still worry that there's something funny in the autostart code, eg: #1152.  Could 4c3e97a have introduced a partial regression?

Not likely, since this just moved some DB setup code around and didn't touch autostart.

> QImage::scaleHeight: Image is a null image  …repeats another 16 times before/after each unit test

I know. It needs `libqt5svg5` as .deb dependency. My Salsa PR added that. Maybe good to add as general Debian dependency. Since we have many SVG icons.

**Comment by m3nu** [comment]
> For the Debian side of things, solving the failure of `test_archive_mount`, when the `os` library is imported in autostart.py, is needed. See today's new commits on debian/master to reproduce (HEAD @commit:0180782).

This commit is building fine here. Ran `sbuild` 2x. Here the `.build` file: [vorta_0.8.2-1_amd64.build.gz](https://github.com/borgbase/vorta/files/7939492/vorta_0.8.2-1_amd64.build.gz)

**Comment by sten0** [comment]
Manu ***@***.***> writes:

>> QImage::scaleHeight: Image is a null image  …repeats another 16 times before/after each unit test
>
> I know. It needs `libqt5svg5` as .deb dependency. My Salsa PR added that. Maybe good to add as general Debian dependency. Since we have many SVG icons.
>

Aha, so this one was my fault!  Sorry.  Yes, I now also remember you
mentioned this long ago.

>> For the Debian side of things, solving the failure of `test_archive_mount`, when the `os` library is imported in autostart.py, is needed. See today's new commits on debian/master to reproduce (HEAD @commit:0180782).
>
> This commit is building fine here. Ran `sbuild` 2x. Here the `.build` file: [vorta_0.8.2-1_amd64.build.gz](https://github.com/borgbase/vorta/files/7939492/vorta_0.8.2-1_amd64.build.gz)
>  

Thanks for the build log, much appreciated.  I've also confirmed that
adding libqt5svg5 solves this test failure on my build system, even
though it doesn't seem to be necessary on yours, which is surprising and
feels like voodoo...  I'm confident we've run into at least one or two
yet unidentified bug in the course of preparing 0.8.2/0.8.3, but not
necessarily and obviously in Vorta--it could be a PyQt bug or
elsewhere.  Hopefully a reproducible trigger will emerge during
real-world testing...

When submitting salsa MRs, please make commits atomic per
operation/topic.  eg: Don't add a dependency in a commit called "Remove
outdated dependencies".  It makes the history hard to follow, especially
since there should be a strong link between items in debian/changelog
and git history.  No need to write changelog entries unless you want to
learn about `dch`, I'm happy to create your section from commit
messages.

On that topic, did you want to submit an MR (oh yeah, that's GitLab
speak for PR) for the libqt5svg5 fix, or shall I go ahead and add it,
along with the conventional "Thanks to Manuel Riel" courtesy message in
the changelog?

**Comment by m3nu** [comment]
> When submitting salsa MRs, please make commits atomic per operation/topic.

Yes, I realize the benefits of this now. 🙈

> On that topic, did you want to submit an MR (oh yeah, that's GitLab speak for PR) for the libqt5svg5 fix

I'll to the MR myself to practice a bit.

> I've also confirmed that adding libqt5svg5 solves this test failure on my build system

Not surprised. When icons are missing, it could cause other issues like preventing clicks on an element.

**Comment by sten0** [comment]
Manu ***@***.***> writes:

>> When submitting salsa MRs, please make commits atomic per operation/topic.
>
> Yes, I realize the benefits of this now. 🙈
>

Welcome to the light! :p

>> On that topic, did you want to submit an MR (oh yeah, that's GitLab speak for PR) for the libqt5svg5 fix
>
> I'll to the MR myself to practice a bit.
>

:-)

>> I've also confirmed that adding libqt5svg5 solves this test failure on my build system
>
> Not surprised. When icons are missing, it could cause other issues like preventing clicks on an element.
>

Now it's my turn to 🙈  Now that you've said it, this is totally obvious.
Yes, this is the first time I'm worked with CI that tested clickable
elements.  Sorry for my ignorance, and thank you for your patience and
understanding.

**Comment by m3nu** [comment]
No worries. Learning every days.

I've now added 2 MRs. One for SVG and one to housekeep dependencies: https://salsa.debian.org/python-team/packages/vorta/-/merge_requests

**Comment by sandrotosi** [comment]
@m3nu can you tag this issue with any of these labels https://github.com/borgbase/vorta/blob/master/.github/stale.yml#L8-L18 so this issue wont be closed? thanks!

**Comment by m3nu** [comment]
Is this still an issue? By now we have a short timeout and kill the whole subprocess group after that.

**Comment by sandrotosi** [comment]
oh i tried with 0.8.2 and the underlying borg process gets killed when cancelling a backup, this can be closed -- thanks!

**Comment by m3nu** [comment]
Happy to hear. Thanks for reporting and confirming the solution!

**Comment by sten0** [comment]
Sandro Tosi ***@***.***> writes:

> @m3nu can you tag this issue with any of these labels https://github.com/borgbase/vorta/blob/master/.github/stale.yml#L8-L18 so this issue wont be closed? thanks!
>

+1

Oh, and here are upstream Borg discussions related to handling lock
cleanups during SIGINT.

https://github.com/borgbackup/borg/issues/519
https://github.com/borgbackup/borg/issues/562
https://github.com/borgbackup/borg/pull/1246

I think that Borg ≥ 1.1.16 will do the right thing if Vorta sends a
SIGINT to cancel the archive check, but I'm not 100% sure.
@ThomasWaldmann would you please ACK||NACK about whether this is safe?

**Comment by ThomasWaldmann** [comment]
Yes, sending SIGINT should would ok with any recent borg version.

Some very recent changes even made some more operations save progress cleanly so the next invocation works faster. Also I fixed that Ctrl-C does not kill the ssh subprocess of borg. This is when pressing Ctrl-C **once**.

When vorta would send SIGINT, it should send it only once and only to the borg process (not to the ssh subprocess).

A second Ctrl-C will kill it quicker / less clean, but nothing really bad should happen then either.
